diff --git a/dist/cjs/CatchBoundary.cjs b/dist/cjs/CatchBoundary.cjs
index ff4a0c0742ea0ad03b0f1c3799876c6a101ebbb0..cfd95bb0657aa43f36949a2ec6fe224aa2dbd289 100644
--- a/dist/cjs/CatchBoundary.cjs
+++ b/dist/cjs/CatchBoundary.cjs
@@ -25,7 +25,7 @@ var CatchBoundaryImpl = class extends react.Component {
 		return { resetKey };
 	}
 	static getDerivedStateFromError(error) {
-		return { error };
+		return { error: error || new Error(`Unhandled falsy error: ${String(error)}`) };
 	}
 	componentDidCatch(error, errorInfo) {
 		this.props.onCatch?.(error, errorInfo);
diff --git a/dist/esm/CatchBoundary.js b/dist/esm/CatchBoundary.js
index af07b6cc914f6025add8a3c95c3fb41af4f6de8a..b29859a38515964d06860d9b11efe2587b05a673 100644
--- a/dist/esm/CatchBoundary.js
+++ b/dist/esm/CatchBoundary.js
@@ -23,7 +23,7 @@ var CatchBoundaryImpl = class extends React$1.Component {
 		return { resetKey };
 	}
 	static getDerivedStateFromError(error) {
-		return { error };
+		return { error: error || new Error(`Unhandled falsy error: ${String(error)}`) };
 	}
 	componentDidCatch(error, errorInfo) {
 		this.props.onCatch?.(error, errorInfo);
