diff --git a/dist/cjs/CatchBoundary.cjs b/dist/cjs/CatchBoundary.cjs
index 6b5625ceb9aec435b99b38451387289ea76882b2..3b1da10a55e584bafd5e558bcad01611c8fa584f 100644
--- a/dist/cjs/CatchBoundary.cjs
+++ b/dist/cjs/CatchBoundary.cjs
@@ -22,7 +22,7 @@ var CatchBoundary = class extends react.Component {
 		return { resetKey };
 	}
 	static getDerivedStateFromError(error) {
-		return { error };
+		return { error: error || new Error(`Unhandled falsy error: ${String(error)}`) };
 	}
 	componentDidCatch(error, errorInfo) {
 		this.props.onCatch?.(error, errorInfo);
diff --git a/dist/esm/CatchBoundary.js b/dist/esm/CatchBoundary.js
index ad786b53f499769176aa510632f0872fbaa3c7fb..db48029566226a1e7807ee0dc8e3acad69187fdc 100644
--- a/dist/esm/CatchBoundary.js
+++ b/dist/esm/CatchBoundary.js
@@ -20,7 +20,7 @@ var CatchBoundary = class extends React$1.Component {
 		return { resetKey };
 	}
 	static getDerivedStateFromError(error) {
-		return { error };
+		return { error: error || new Error(`Unhandled falsy error: ${String(error)}`) };
 	}
 	componentDidCatch(error, errorInfo) {
 		this.props.onCatch?.(error, errorInfo);
